File: Napakalakirb/lib/death_bad_consequence.rb
#encoding:utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

class DeathBadConsequence < NumericBadConsequence
  
  attr_reader :death
  
  def initialize(aText)
    super(aText,Player::MAXLEVEL,MAXTREASURES,MAXTREASURES)
    @death = true;
  end
  
  def isEmpty
    return super()
  end
  
  def adjustToFitTreasureLists( v , h )
    return super(v,h)
  end
  
  def to_s
    return super() + ", Muerte: #{@death}"
  end
  
end

File: Napakalaki/src/napakalaki/DeathBadConsequence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;

import java.util.ArrayList;

/**
 *
 * @author jose
 */
public class DeathBadConsequence extends NumericBadConsequence {
    
    private boolean death;
    
    public DeathBadConsequence( String text, boolean death){
        super(text,Player.MAXLEVEL,BadConsequence.getMaxTreasures(),BadConsequence.getMaxTreasures());
        this.death = death;
    }
    
    @Override
    public boolean isEmpty(){
        return super.isEmpty();
    }
    
    public boolean getDeath(){
        return death;
    }
    
    @Override
    public String toString(){
        return super.toString() + " Muerto: " + death;
    }
    
}
